tests/e2e/system_test/test_cluster_full_and_recovery.py

```diff
@@ -0,0 +1,348 @@
+import logging
+import time
+
+from ocs_ci.helpers import helpers
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.ocs.benchmark_operator_fio import get_file_size
+from ocs_ci.framework.pytest_customization.marks import system_test
+from ocs_ci.utility.prometheus import PrometheusAPI
+from ocs_ci.utility.utils import TimeoutSampler, ceph_health_check
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.benchmark_operator_fio import BenchmarkOperatorFIO
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.ocs.resources.pod import cal_md5sum
+from ocs_ci.helpers import disruption_helpers
+from ocs_ci.ocs.cluster import (
+    change_ceph_backfillfull_ratio,
+    change_ceph_full_ratio,
+    get_percent_used_capacity,
+)
+
+log = logging.getLogger(__name__)
+
+
+@system_test
+class TestClusterFullAndRecovery(E2ETest):
+    """
+    Test Cluster Full And Recovery
+
+    """
+
+    def teardown(self):
+        if self.banchmark_operator_teardown:
+            change_ceph_full_ratio(95)
+            self.benchmark_obj.cleanup()
+            change_ceph_backfillfull_ratio(95)
+            ceph_health_check(tries=30, delay=60)
+        change_ceph_backfillfull_ratio(80)
+        change_ceph_full_ratio(85)
+
+    def test_cluster_full_and_recovery(
+        self,
+        teardown_project_factory,
+        snapshot_restore_factory,
+        snapshot_factory,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        1.Create PVC1 [FS + RBD]
+        2.Verify new PVC1 [FS + RBD] on Bound state
+        3.Run FIO on PVC1_FS + PVC1_RBD
+        4.Calculate Checksum PVC1_FS + PVC1_RBD
+        5.Fill the cluster to “Full ratio” (usually 85%) with benchmark-operator
+        6.Verify Alerts are seen ["CephClusterCriticallyFull", "CephOSDNearFull"]
+        7.Respin 'osd' ,'mgr', 'mon' pods
+        8.Create PVC2 [FS + RBD]
+        9.Verify PVC2 [FS + RBD] are in Pending state
```

Comment:
We are handling both Cephfs RBD and filesystem in the same test, should we parametrize the test based on the interface?

Reply:
It takes a long time. So I combined ceph-fs + ceph-rbd  on same test